pkg/proxy/proxy_test.go

```diff
@@ -73,11 +74,12 @@ func (s *ProxySuite) TestPortAllocator(c *C) {
 	c.Assert(pp.rulesPort, Equals, uint16(0))
 
 	// Allocates a different port (due to port was never acked)
-	port2, err := p.AllocateProxyPort("listener1", true)
+	port2, err := p.AllocateProxyPort("listener1", true, false)
 	c.Assert(err, IsNil)
 	c.Assert(port2, Not(Equals), port)
 	c.Assert(pp.proxyType, Equals, ProxyTypeCRD)
 	c.Assert(pp.ingress, Equals, false)
+	c.Assert(pp.localOnly, Equals, true)
```

Comment:
This seems weird, even though this is similar how the `ingress` parameter is handled in this case. This test would be more readable if we re-set both `ingress` and `localOnly` members when (re-)allocating the `proxyPort` after it was seen as zero in `AllocateProxyPort()`.

Reply:
https://github.com/cilium/cilium/issues/24264